### ISSUE TITLE
How to define an abstract state with ng-forward?...

### ISSUE THREAD

**Issue by borntorun** [brandonroberts/ng-forward-ui-router-example#1]
Hi,

I'm trying using ng-forward with ui-router and I come to your example... and because they refer this example as a reference to use ui-router, I'm putting this question here.

I'm trying to use it on an app that also also 'ui-router-extras' and its features like 'Deep State Redirect' and 'Sticky states'...

I'm trying to set something like this state tree:
- A 
  - A.child
    - A.child.child  

Where A is my root state which is an abstract state 
`$stateProvider.state('A', { .... abstract: true })`

**A.child** is triggered by default:

```
$stateProvider.state('A', {
      abstract: true,
      url: '/',
      deepStateRedirect: {
        default: {
          state: 'A.child'
        }
      }
      ....
    })
```

**A.child** has also childs which one of them is triggered by default...

**A.child** also define some views to be set

So, resuming...

How I define an abstract state using ng-forward? it is possible, or I must do it the "angular1 way"?

Thanks

**Comment by borntorun** [comment]
Well... my question is not really related on how to create an abastract state but how to handle the parent-child load and deepstateredirect with ui-router and ui-router-extras? 
Any ideas?

**Comment by brandonroberts** [comment]
You should be able to add any ui-router specific properties to your state objects in your StateConfig.

``` js
@StateConfig([{
      name: 'A',
      abstract: true,
      url: '/',
      deepStateRedirect: {
        default: {
          state: 'A.child'
        }
      }
      ....
}])
```

**Comment by borntorun** [comment]
Yes I did just that...or similar...but the child state is not triggered.
(I realized after bootstrap an abstract state already exists so I'm not defining an abstract)

What I think: when using @StateConfig we are postponing the state configuration for the moment when the child component is loaded. So 'FutureState' feature from ui-router-extras is needed for configuring those future states. But 'FutureState' need to lazy load the states from a "module" and I can't find a way to do that with ng-forward. 
I tried to create a angular 1 bundle from my component (which is I think is stupid...), the module was loade and registered, but then in the moment of state transition the child state is not recognized, because in fact, it is not being configured in the bundle module but instead in the @StateConfig decorator of the component. I can not connect the two parts: @StateConfig and FutureState+Module configuration.

This is very tricky. My intention was to use ng-forward to begin my upgrade path to angular 2 but it seems that with not so trivial apps/examples maybe is not the way to go...

**Comment by brandonroberts** [comment]
It seems like your use case is more advanced than what the @StateConfig decorator will allow you to do. You could always create a separate module and attach a `.config` to it to do something more advanced with ui-router, while still keeping the benefits of ng-forward.

**Comment by borntorun** [comment]
Well, I continued thinking about and I tried to define all my states in the component that I use to bootstrap the app; and that's is the only way I achieve to go into the child state. But I can´t get its template to show. 
One thing that makes me confusing is why @StateConfig is a component decorator? I think it should not be. In angular 1 a state is like a place-in-time of the application and several components can be triggered by the state.
Having helpers like ng-forward to mimic the angular2 syntax in a angular1 app is great and a good idea. But have to mix the two syntaxes in a application hummm... seems cumbersome and not so much productive...

**Comment by brandonroberts** [comment]
Agreed. The way I think about it is, ng-forward is a way to shift your mental thinking more towards how things work in Angular 2 and providing you a head start. There will always be some areas that aren't covered by ng-forward and that's somewhat intentional. Hopefully at the end, you will have less things to migrate over to Angular 2 and there will be ways to accomplish those things that were left over.
